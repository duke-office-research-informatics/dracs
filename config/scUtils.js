import React from "react";
import { ThemeProvider } from "styled-components";
import { mount, shallow } from "enzyme";
import { theme as defaultTheme } from "../lib/dracs.es.js";

export const shallowWithTheme = (children, customTheme) => {
  const theme = customTheme || defaultTheme;
  return shallow(children, { context: theme });
};

/* eslint-disable */
export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}

//FOR ACCESSING PORTALS VIA ENZYME -- import portal from dracs
// https://github.com/airbnb/enzyme/issues/1202#issuecomment-351639783
// function mountPortal(el){
//   const wrapper =  mountWithTheme(el);
//   wrapper.setState({ visible: true});
//   const portalInstance = wrapper.find(Portal).instance();
//   return mountWithTheme(portalInstance.props.children);
// }

//FOR ACCESSING EVENT LISTENERS ATTACHED TO DOCUMENT OR DOCUMENT.BODY
//https://github.com/airbnb/enzyme/issues/426#issuecomment-296304015
// const map = {};
// document.body.addEventListener = jest.fn((event, cb) => {
//   map[event] = cb;
// });
