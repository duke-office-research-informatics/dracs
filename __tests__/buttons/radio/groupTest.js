import React from "react";
import { mountWithTheme } from "../../../config/scUtils.js";
import { axe } from "../../../config/axe-helper.js";
import { RadioGroup, RadioButton, theme } from "../../../lib/dracs.es.js";

const mockFn = jest.fn();

describe("RadioGroup", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mountWithTheme(
      <RadioGroup name="testGroup" onChange={mockFn} value="button1">
        <RadioButton label="button 1" name="button1" value="button1" />
        <RadioButton label="button 2" name="button2" value="button2" />
      </RadioGroup>
    );
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("returns no a11y violations", async () => {
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });

  it("calls the change handler when a radio button is clicked", () => {
    const btn = wrapper
      .find(RadioButton)
      .last()
      .find("buttons__RadioInput");
    btn.simulate("click");
    expect(mockFn).toHaveBeenCalled();
  });

  it("sets buttons as disabled when `disabled` is set to true", () => {
    wrapper.setProps({ disabled: true });
    expect(wrapper).toMatchSnapshot();
    const btn1 = wrapper
      .find(RadioButton)
      .first()
      .find("buttons__RadioInput");
    const btn2 = wrapper
      .find(RadioButton)
      .last()
      .find("buttons__RadioInput");
    expect(btn1.getDOMNode().attributes.getNamedItem("disabled").value).toBe(
      ""
    );
    expect(btn2.getDOMNode().attributes.getNamedItem("disabled").value).toBe(
      ""
    );
  });

  it("applies styles to the wrapper via `wrapperStyle` prop", () => {
    wrapper.setProps({ wrapperStyle: { color: theme.colors.action } });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("group__RadioGroupWrap").props().style).toHaveProperty(
      "color",
      theme.colors.action
    );
  });
});
