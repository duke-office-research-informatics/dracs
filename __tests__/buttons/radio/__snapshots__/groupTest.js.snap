// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup applies styles to the wrapper via \`wrapperStyle\` prop 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  min-height: 24px;
  min-width: 24px;
  margin: 4px;
}

.c1 input[type="radio"] {
  opacity: 0;
}

.c1 input[type="radio"]:focus + label::before {
  outline-offset: 2px;
  outline: dotted 1px rgb(59,153,252);
}

.c1 input[type="radio"] + label {
  position: absolute;
  left: 0px;
  top: 0px;
  display: inline-block;
  cursor: pointer;
}

.c1 input[type="radio"]:checked + label:before {
  content: " ";
  display: inline-block;
  height: 24px;
  width: 24px;
  background: url(data:image/svg+xml;base64,PHN2ZyBmaWxsPSIjMDY4MGNkIiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxwYXRoIGQ9Ik0xMiA3Yy0yLjc2IDAtNSAyLjI0LTUgNXMyLjI0IDUgNSA1IDUtMi4yNCA1LTUtMi4yNC01LTUtNXptMC01QzYuNDggMiAyIDYuNDggMiAxMnM0LjQ4IDEwIDEwIDEwIDEwLTQuNDggMTAtMTBTMTcuNTIgMiAxMiAyem0wIDE4Yy00LjQyIDAtOC0zLjU4LTgtOHMzLjU4LTggOC04IDggMy41OCA4IDgtMy41OCA4LTggOHoiLz4KICAgIDxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz4KPC9zdmc+);
}

.c1 input[type="radio"] + label:before {
  content: " ";
  display: inline-block;
  height: 24px;
  width: 24px;
  background: url(data:image/svg+xml;base64,PHN2ZyBmaWxsPSIjMDY4MGNkIiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxwYXRoIGQ9Ik0xMiAyQzYuNDggMiAyIDYuNDggMiAxMnM0LjQ4IDEwIDEwIDEwIDEwLTQuNDggMTAtMTBTMTcuNTIgMiAxMiAyem0wIDE4Yy00LjQyIDAtOC0zLjU4LTgtOHMzLjU4LTggOC04IDggMy41OCA4IDgtMy41OCA4LTggOHoiLz4KICAgIDxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz4KPC9zdmc+);
}

.c1 input[type="radio"]:checked:disabled + label:before {
  content: " ";
  display: inline-block;
  height: 24px;
  width: 24px;
  background: url(data:image/svg+xml;base64,PHN2ZyBmaWxsPSIjYjViNWI1IiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxwYXRoIGQ9Ik0xMiA3Yy0yLjc2IDAtNSAyLjI0LTUgNXMyLjI0IDUgNSA1IDUtMi4yNCA1LTUtMi4yNC01LTUtNXptMC01QzYuNDggMiAyIDYuNDggMiAxMnM0LjQ4IDEwIDEwIDEwIDEwLTQuNDggMTAtMTBTMTcuNTIgMiAxMiAyem0wIDE4Yy00LjQyIDAtOC0zLjU4LTgtOHMzLjU4LTggOC04IDggMy41OCA4IDgtMy41OCA4LTggOHoiLz4KICAgIDxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz4KPC9zdmc+);
  cursor: not-allowed;
}

.c1 input[type="radio"]:not(:checked):disabled + label:before {
  content: " ";
  display: inline-block;
  height: 24px;
  width: 24px;
  background: url(data:image/svg+xml;base64,PHN2ZyBmaWxsPSIjYjViNWI1IiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxwYXRoIGQ9Ik0xMiAyQzYuNDggMiAyIDYuNDggMiAxMnM0LjQ4IDEwIDEwIDEwIDEwLTQuNDggMTAtMTBTMTcuNTIgMiAxMiAyem0wIDE4Yy00LjQyIDAtOC0zLjU4LTgtOHMzLjU4LTggOC04IDggMy41OCA4IDgtMy41OCA4LTggOHoiLz4KICAgIDxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz4KPC9zdmc+);
  cursor: not-allowed;
}

.c1 span {
  margin-left: 24px;
  font-size: 1em;
  color: #131516;
}

<div
  aria-activedescendant="button1"
  className="c0"
  name="testGroup"
  style={
    Object {
      "color": "#0680cd",
    }
  }
>
  <div
    className="c1"
  >
    <input
      aria-checked={true}
      aria-labelledby="button1-label"
      checked={true}
      className=""
      disabled={false}
      id="button1"
      name="button1"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
      value="button1"
    />
    <label
      htmlFor="button1"
    />
    <span
      id="button1-label"
      onClick={null}
    >
      button 1
    </span>
  </div>
  <div
    className="c1"
  >
    <input
      aria-checked={false}
      aria-labelledby="button2-label"
      checked={false}
      className=""
      disabled={false}
      id="button2"
      name="button2"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
      value="button2"
    />
    <label
      htmlFor="button2"
    />
    <span
      id="button2-label"
      onClick={null}
    >
      button 2
    </span>
  </div>
</div>
`;

exports[`RadioGroup matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  min-height: 24px;
  min-width: 24px;
  margin: 4px;
}

.c1 input[type="radio"] {
  opacity: 0;
}

.c1 input[type="radio"]:focus + label::before {
  outline-offset: 2px;
  outline: dotted 1px rgb(59,153,252);
}

.c1 input[type="radio"] + label {
  position: absolute;
  left: 0px;
  top: 0px;
  display: inline-block;
  cursor: pointer;
}

.c1 input[type="radio"]:checked + label:before {
  content: " ";
  display: inline-block;
  height: 24px;
  width: 24px;
  background: url(data:image/svg+xml;base64,PHN2ZyBmaWxsPSIjMDY4MGNkIiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxwYXRoIGQ9Ik0xMiA3Yy0yLjc2IDAtNSAyLjI0LTUgNXMyLjI0IDUgNSA1IDUtMi4yNCA1LTUtMi4yNC01LTUtNXptMC01QzYuNDggMiAyIDYuNDggMiAxMnM0LjQ4IDEwIDEwIDEwIDEwLTQuNDggMTAtMTBTMTcuNTIgMiAxMiAyem0wIDE4Yy00LjQyIDAtOC0zLjU4LTgtOHMzLjU4LTggOC04IDggMy41OCA4IDgtMy41OCA4LTggOHoiLz4KICAgIDxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz4KPC9zdmc+);
}

.c1 input[type="radio"] + label:before {
  content: " ";
  display: inline-block;
  height: 24px;
  width: 24px;
  background: url(data:image/svg+xml;base64,PHN2ZyBmaWxsPSIjMDY4MGNkIiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxwYXRoIGQ9Ik0xMiAyQzYuNDggMiAyIDYuNDggMiAxMnM0LjQ4IDEwIDEwIDEwIDEwLTQuNDggMTAtMTBTMTcuNTIgMiAxMiAyem0wIDE4Yy00LjQyIDAtOC0zLjU4LTgtOHMzLjU4LTggOC04IDggMy41OCA4IDgtMy41OCA4LTggOHoiLz4KICAgIDxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz4KPC9zdmc+);
}

.c1 input[type="radio"]:checked:disabled + label:before {
  content: " ";
  display: inline-block;
  height: 24px;
  width: 24px;
  background: url(data:image/svg+xml;base64,PHN2ZyBmaWxsPSIjYjViNWI1IiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxwYXRoIGQ9Ik0xMiA3Yy0yLjc2IDAtNSAyLjI0LTUgNXMyLjI0IDUgNSA1IDUtMi4yNCA1LTUtMi4yNC01LTUtNXptMC01QzYuNDggMiAyIDYuNDggMiAxMnM0LjQ4IDEwIDEwIDEwIDEwLTQuNDggMTAtMTBTMTcuNTIgMiAxMiAyem0wIDE4Yy00LjQyIDAtOC0zLjU4LTgtOHMzLjU4LTggOC04IDggMy41OCA4IDgtMy41OCA4LTggOHoiLz4KICAgIDxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz4KPC9zdmc+);
  cursor: not-allowed;
}

.c1 input[type="radio"]:not(:checked):disabled + label:before {
  content: " ";
  display: inline-block;
  height: 24px;
  width: 24px;
  background: url(data:image/svg+xml;base64,PHN2ZyBmaWxsPSIjYjViNWI1IiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxwYXRoIGQ9Ik0xMiAyQzYuNDggMiAyIDYuNDggMiAxMnM0LjQ4IDEwIDEwIDEwIDEwLTQuNDggMTAtMTBTMTcuNTIgMiAxMiAyem0wIDE4Yy00LjQyIDAtOC0zLjU4LTgtOHMzLjU4LTggOC04IDggMy41OCA4IDgtMy41OCA4LTggOHoiLz4KICAgIDxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz4KPC9zdmc+);
  cursor: not-allowed;
}

.c1 span {
  margin-left: 24px;
  font-size: 1em;
  color: #131516;
}

<div
  aria-activedescendant="button1"
  className="c0"
  name="testGroup"
>
  <div
    className="c1"
  >
    <input
      aria-checked={true}
      aria-labelledby="button1-label"
      checked={true}
      className=""
      disabled={false}
      id="button1"
      name="button1"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
      value="button1"
    />
    <label
      htmlFor="button1"
    />
    <span
      id="button1-label"
      onClick={null}
    >
      button 1
    </span>
  </div>
  <div
    className="c1"
  >
    <input
      aria-checked={false}
      aria-labelledby="button2-label"
      checked={false}
      className=""
      disabled={false}
      id="button2"
      name="button2"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
      value="button2"
    />
    <label
      htmlFor="button2"
    />
    <span
      id="button2-label"
      onClick={null}
    >
      button 2
    </span>
  </div>
</div>
`;

exports[`RadioGroup sets buttons as disabled when \`disabled\` is set to true 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  min-height: 24px;
  min-width: 24px;
  margin: 4px;
}

.c1 input[type="radio"] {
  opacity: 0;
}

.c1 input[type="radio"]:focus + label::before {
  outline-offset: 2px;
  outline: dotted 1px rgb(59,153,252);
}

.c1 input[type="radio"] + label {
  position: absolute;
  left: 0px;
  top: 0px;
  display: inline-block;
  cursor: pointer;
}

.c1 input[type="radio"]:checked + label:before {
  content: " ";
  display: inline-block;
  height: 24px;
  width: 24px;
  background: url(data:image/svg+xml;base64,PHN2ZyBmaWxsPSIjMDY4MGNkIiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxwYXRoIGQ9Ik0xMiA3Yy0yLjc2IDAtNSAyLjI0LTUgNXMyLjI0IDUgNSA1IDUtMi4yNCA1LTUtMi4yNC01LTUtNXptMC01QzYuNDggMiAyIDYuNDggMiAxMnM0LjQ4IDEwIDEwIDEwIDEwLTQuNDggMTAtMTBTMTcuNTIgMiAxMiAyem0wIDE4Yy00LjQyIDAtOC0zLjU4LTgtOHMzLjU4LTggOC04IDggMy41OCA4IDgtMy41OCA4LTggOHoiLz4KICAgIDxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz4KPC9zdmc+);
}

.c1 input[type="radio"] + label:before {
  content: " ";
  display: inline-block;
  height: 24px;
  width: 24px;
  background: url(data:image/svg+xml;base64,PHN2ZyBmaWxsPSIjMDY4MGNkIiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxwYXRoIGQ9Ik0xMiAyQzYuNDggMiAyIDYuNDggMiAxMnM0LjQ4IDEwIDEwIDEwIDEwLTQuNDggMTAtMTBTMTcuNTIgMiAxMiAyem0wIDE4Yy00LjQyIDAtOC0zLjU4LTgtOHMzLjU4LTggOC04IDggMy41OCA4IDgtMy41OCA4LTggOHoiLz4KICAgIDxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz4KPC9zdmc+);
}

.c1 input[type="radio"]:checked:disabled + label:before {
  content: " ";
  display: inline-block;
  height: 24px;
  width: 24px;
  background: url(data:image/svg+xml;base64,PHN2ZyBmaWxsPSIjYjViNWI1IiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxwYXRoIGQ9Ik0xMiA3Yy0yLjc2IDAtNSAyLjI0LTUgNXMyLjI0IDUgNSA1IDUtMi4yNCA1LTUtMi4yNC01LTUtNXptMC01QzYuNDggMiAyIDYuNDggMiAxMnM0LjQ4IDEwIDEwIDEwIDEwLTQuNDggMTAtMTBTMTcuNTIgMiAxMiAyem0wIDE4Yy00LjQyIDAtOC0zLjU4LTgtOHMzLjU4LTggOC04IDggMy41OCA4IDgtMy41OCA4LTggOHoiLz4KICAgIDxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz4KPC9zdmc+);
  cursor: not-allowed;
}

.c1 input[type="radio"]:not(:checked):disabled + label:before {
  content: " ";
  display: inline-block;
  height: 24px;
  width: 24px;
  background: url(data:image/svg+xml;base64,PHN2ZyBmaWxsPSIjYjViNWI1IiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxwYXRoIGQ9Ik0xMiAyQzYuNDggMiAyIDYuNDggMiAxMnM0LjQ4IDEwIDEwIDEwIDEwLTQuNDggMTAtMTBTMTcuNTIgMiAxMiAyem0wIDE4Yy00LjQyIDAtOC0zLjU4LTgtOHMzLjU4LTggOC04IDggMy41OCA4IDgtMy41OCA4LTggOHoiLz4KICAgIDxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz4KPC9zdmc+);
  cursor: not-allowed;
}

.c1 span {
  margin-left: 24px;
  font-size: 1em;
  color: #696969;
  cursor: not-allowed;
}

<div
  aria-activedescendant="button1"
  className="c0"
  name="testGroup"
>
  <div
    className="c1"
    disabled={true}
  >
    <input
      aria-checked={true}
      aria-labelledby="button1-label"
      checked={true}
      className=""
      disabled={true}
      id="button1"
      name="button1"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
      value="button1"
    />
    <label
      htmlFor="button1"
    />
    <span
      disabled={true}
      id="button1-label"
      onClick={null}
    >
      button 1
    </span>
  </div>
  <div
    className="c1"
    disabled={true}
  >
    <input
      aria-checked={false}
      aria-labelledby="button2-label"
      checked={false}
      className=""
      disabled={true}
      id="button2"
      name="button2"
      onChange={[Function]}
      onClick={[Function]}
      type="radio"
      value="button2"
    />
    <label
      htmlFor="button2"
    />
    <span
      disabled={true}
      id="button2-label"
      onClick={null}
    >
      button 2
    </span>
  </div>
</div>
`;
